src/angular-datatables.directive.ts

```diff
@@ -62,62 +63,60 @@ export class DataTableDirective implements OnDestroy, OnInit {
-                  // get <td> element which holds data using index
-                  const rowFromCol = row.childNodes.item(i);
-                  // Transform data with Pipe
-                  const rowVal = $(rowFromCol).text();
-                  const rowValAfter = pipe.transform(rowVal);
-                  // Apply transformed string to <td>
-                  $(rowFromCol).text(rowValAfter);
-                });
+    // resolve dtOptions if Promise was provided.
+    if ('then' in this.dtOptions) this.dtOptions = await Promise.resolve(this.dtOptions);
+    // process request
+    this.dtInstance = this.ngZone.runTask(_ => {
+      // Assign DT properties here
+      let options: ADTSettings = {
+        rowCallback: (row, data, index) => {
+          if (this.dtOptions.columns) {
+            const columns = this.dtOptions.columns;
+            // Filter columns with pipe declared
+            const colsWithPipe = columns.filter(x => x.ngPipeInstance && !x.ngTemplateRef);
+            // Iterate
+            colsWithPipe.forEach(el => {
+              const pipe = el.ngPipeInstance;
+              // find index of column using `data` attr
+              const i = columns.findIndex(e => e.data == el.data);
+              // get <td> element which holds data using index
+              const rowFromCol = row.childNodes.item(i);
+              // Transform data with Pipe
+              const rowVal = $(rowFromCol).text();
+              const rowValAfter = pipe.transform(rowVal);
+              // Apply transformed string to <td>
+              $(rowFromCol).text(rowValAfter);
+            });
 
-                // Filter columns using `ngTemplateRef`
-                const colsWithTemplate = columns.filter(x => x.ngTemplateRef && !x.ngPipeInstance);
-                colsWithTemplate.forEach(el => {
-                  const { ref, context } = el.ngTemplateRef;
-                  // get <td> element which holds data using index
-                  const index = columns.findIndex(e => e.data == el.data);
-                  const cellFromIndex = row.childNodes.item(index);
-                  // render onto DOM
-                  // finalize context to be sent to user
-                  const _context = Object.assign({}, context, context?.userData, {
-                    adtData: data
-                  });
-                  const instance = self.vcr.createEmbeddedView(ref, _context);
-                  self.renderer.appendChild(cellFromIndex, instance.rootNodes[0]);
-                });
-              }
+            // Filter columns using `ngTemplateRef`
+            const colsWithTemplate = columns.filter(x => x.ngTemplateRef && !x.ngPipeInstance);
+            colsWithTemplate.forEach(el => {
+              const { ref, context } = el.ngTemplateRef;
+              // get <td> element which holds data using index
+              const index = columns.findIndex(e => e.data == el.data);
```

Comment:
__chore__: This variable `index` is shadowing the previous variable `index`. We need to rename it. Moreover, the equality must be `===` instead.

```suggestion
              const columnIndex = columns.findIndex(e => e.data === el.data);
```